### PR TITLE
fix check_hostname needs server_hostname argument by @plockaby

### PR DESCRIPTION
**Pull request by temoto**
Fixes https://github.com/eventlet/eventlet/issues/567

Moved from https://github.com/eventlet/eventlet/pull/575

### CODE REVIEW

File: tests/ssl_test.py [hunk 1]
```diff
@@ -371,3 +372,24 @@ def client(addr):
         peer, _ = server_tls.accept()
         assert peer.recv(64) == expected
         peer.close()
+
+    @tests.skip_if(sys.version_info[:2] < (3, 7))
```
Comment (tipabu) [inline tests/ssl_test.py:376]: Why only run the test on py37+? Looks like the issue was reported on py36...

File: tests/ssl_test.py [hunk 2]
```diff
@@ -371,3 +372,24 @@ def client(addr):
...
+            sock, addr = listener.accept()
+            sock.recv(64)
+            sock.sendall(b"response")
+            sock.close()
+
+        listener = listen_ssl_socket()
+        server_coro = eventlet.spawn(serve, listener)
+        client = ssl.wrap_socket(
+            eventlet.connect(listener.getsockname()),
+            check_hostname=True,
```
Comment (tipabu) [inline tests/ssl_test.py:390]: This doesn't seem to be how CPython's `ssl.wrap_socket()` works:
``` python
>>> import socket, ssl
>>> sock = socket.socket()
>>> sock.connect(('github.com', 443))
>>> ssock = ssl.wrap_socket(sock, check_hostname=True, verify_mode=ssl.CERT_REQUIRED)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: wrap_socket() got an unexpected keyword argument 'check_hostname'
```
WDYT about something like
``` diff
diff --git a/tests/ssl_test.py b/tests/ssl_test.py
index c5b5b49..dda82ff 100644
--- a/tests/ssl_test.py
+++ b/tests/ssl_test.py
@@ -385,11 +385,14 @@ class SSLTest(tests.LimitedTestCase):
 
         listener = listen_ssl_socket()
         server_coro = eventlet.spawn(serve, listener)
-        client = ssl.wrap_socket(
-            eventlet.connect(listener.getsockname()),
-            check_hostname=True,
-            verify_mode=ssl.CERT_REQUIRED,
-        )
+        ctx = ssl.SSLContext()
+        ctx.verify_mode = ssl.CERT_REQUIRED
+        ctx.check_hostname = True
+        ctx.load_verify_locations(tests.certificate_file)
+        client = ctx.wrap_socket(
+            socket.socket(socket.AF_INET, socket.SOCK_STREAM),
+            server_hostname="Test")
+        client.connect(listener.getsockname())
         client.send(b"check_hostname works")
         client.recv(64)
         server_coro.wait()

```
?

**Review by tipabu — CHANGES_REQUESTED** [review]
I'm not seeing the test pop if I back out the change to eventlet/green/ssl.py -- and I'm kinda surprised the test doesn't pass a `server_hostname` arg anywhere :confused:

The diff I included got me a failing test for py36, but (surprisingly?) it passes on master for py37-39 -- not sure what's up with that.

**Comment by temoto** [comment]
@tipabu thanks. It actually breaks on <3.7, the opposite of my version skipping.

**Pull request merged**